interview-questions/activity-interview-questions.md

```diff
@@ -0,0 +1,70 @@
+# Activity: Interview Questions
+
+## Goal
+
+Our goal is to compile a series of interview question resources and interview questions. By the end of this activity, everyone should have access to some recommended interview questions.
+
+## Preparation
+
+The facilitator should make a spreadsheet that includes everyone's name, and allows students to notate at least two questions. For example:
+
+| Name | Question #1 | Question #2 |
+| ---- | ----------- | ----------- |
+|      |             |
+
+The facilitator should also prepare a way to record ideas in part one. Some recommended methods include:
+
+- The word cloud feature of [Mentimeter](https://www.mentimeter.com/)
+- Taking notes on a shared document
+
+## Activity Instructions
+
+There are two parts to this activity:
+
+1. Crowdsource interview resources
+1. Crowdsource interview questions
+
+### Crowdsource Resources
+
+As a large group, follow directions from the facilitator to create a shared list of interview question resources.
+
+These resources may include:
+
+- Websites made for technical interview preparation
+- Newsletters made for technical interview preparation
+- Recommended books
+
+If it includes practice interview questions, then it's valid!
+
+### Crowdsource Questions
+
+Afterwards, individually find and share practice questions. Everyone must contribute at least _two_ questions to the shared document of practice questions.
+
+- One must be a coding question
+- The second may be a behavioral question
+- There can be overlap or repetition compared to other entries in the document.
+
+For each coding question, include a link to its original source if possible. For example:
+
+| Name         | Question #1                                                                 | Question #2                                                                                                         |
+| ------------ | --------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------- |
+| Eamonn Finch | [Roman Numeral to Integer](https://leetcode.com/problems/roman-to-integer/) | Describe a time when you were working with someone and there was a conflict. What happened, and how did it resolve? |
+
+#### Tips
+
+Begin with looking for questions that are:
+
+- Marked as "easy"
+- Centered around:
+  - Arrays
+  - Strings
+  - Hash Tables
+  - Sorting
+
+### !callout-info
+
+## Your Mileage May Vary
+
+Finding interview questions is an imperfect process. Sometimes, questions aren't appropriate after all. Don't aim for perfection, aim for practice!
```

Comment:
I feel like we can provide a little more framing here. What makes a question potentially inappropriate? How can we extract good lessons from even "inappropriate" questions. Keep the purpose for practice in mind.
```suggestion
Finding interview questions is an imperfect process. A practice question might have a solution that is more complex than would be expected to be developed during an interview. It might have an "optimal" solution that is so "clever" that it can't be applied to any other situations. The solution may require additional reading and research to understand _why_ it works. This is all still good practice!

<br/>

Our objective is exposure. We don't review interview questions hoping that we'll have already seen and solved a _particular_ question that an interviewer might ask. We review interview questions to gain exposure to _general strategies_ and approaches.
```

Reply:
I think you're interpreting an unintended meaning!

Your suggestion seems to be saying "A practice question may be imperfect because of its appropriateness."

I was attempting to make a comment on the literal process of finding them! For example, if I find an interview question, it may sound like a great question, but it ends up being totally inappropriate in skill level, content, or really weirdly worded.

I'll try a counter suggestion:

```suggestion
Online interview prep material can target entry-level developers or senior-level developers. There are interview questions that target general programming, software development, or computer science, and some that target specific subjects, such as networking or databases.

<br/>

Finding interview questions is an imperfect process. Sometimes the question we found targets the wrong skill level or subject material, and we don't realize this until we're practicing the interview. Sometimes, the question we've found is in a different programming language!

<br/>

Even after coming across a few strange interview questions, keep persisting! Getting presented with unlikely problems and learning how to assess an appropriate interview question for you is part of the process. Don't aim for perfection, aim for practice!
```

Reply:
Ooh! I love it! I think this is really clear.

Just a couple tense/wording suggestions on the second paragraph.

```suggestion
Online interview prep material can target entry-level developers or senior-level developers. There are interview questions that target general programming, software development, or computer science, and some that target specific subjects, such as networking or databases.
<br/>
Finding interview questions is an imperfect process. Sometimes a question we find targets the wrong skill level or subject material, and we don't realize this until we're doing our interview practice. Sometimes, a question we find is in a different programming language!
<br/>
Even after coming across a few strange interview questions, keep persisting! Getting presented with unlikely problems and learning how to assess an appropriate interview question for you is part of the process. Don't aim for perfection, aim for practice!
```